src/openforms/payments/contrib/worldline/plugin.py

```diff
@@ -0,0 +1,255 @@
+
+    def handle_return(
+        self,
+        request: Request,
+        payment: SubmissionPayment,
+        options: PaymentOptions,
+    ) -> HttpResponse:
+        with structlog.contextvars.bound_contextvars(
+            submission_uuid=str(payment.submission.uuid),
+            payment_uuid=str(payment.uuid),
+            plugin=self,
+            entrypoint="browser",
+        ):
+            log = logger.bind()
+            log.info("process_payment_status")
+
+            returnmac = payment.plugin_options.get("returnmac", "")
+            checkout_id = payment.plugin_options.get("checkout_id", "")
+
+            if (
+                not request.query_params.get("RETURNMAC") == returnmac
+                or not request.query_params.get("hostedCheckoutId") == checkout_id
+            ):
+                return HttpResponseBadRequest(b"Incorrect query parameters provided")
+
+            token = submission_status_token_generator.make_token(payment.submission)
+            status_url = request.build_absolute_uri(
+                reverse(
+                    "api:submission-status",
+                    kwargs={"uuid": payment.submission.uuid, "token": token},
+                )
+            )
+
+            client = options["merchant"].get_checkout_client()
+
+            try:
+                response = client.get_hosted_checkout(checkout_id)
+            except ApiException:
+                return HttpResponseServerError(b"Unable to retrieve checkout status")
+
+            payment_data = response.created_payment_output
+            checkout_status = response.status if response.status else ""
+            status = (
+                payment_data.payment.status
+                if payment_data and payment_data.payment and payment_data.payment.status
+                else ""
+            )
+
+            payment_provider_id = (
+                payment_data.payment.id
+                if payment_data and payment_data.payment and payment_data.payment.id
+                else ""
+            )
+
+            try:
+                self.apply_status(
+                    payment,
+                    status,
+                    checkout_status,
+                    payment_provider_id,
+                )
+            except KeyError:
+                return HttpResponseServerError(b"Unable to retrieve payment status")
+
+            redirect_url = get_frontend_redirect_url(
+                payment.submission,
+                action="payment",
+                action_params={
+                    "of_payment_status": payment.status,
+                    "of_payment_id": str(payment.uuid),
+                    "of_submission_status": status_url,
+                },
+            )
+            return HttpResponseRedirect(redirect_url)
+
+    # TODO
+    def handle_webhook(self, request: Request) -> SubmissionPayment:
+        raise NotImplementedError()
+
+    @classmethod
+    def iter_config_checks(cls) -> Generator[Entry, Any, Any]:
```

Comment:
```suggestion
    def iter_config_checks(cls) -> Iterator[Entry]:
```
usually this is the more ergonomic variant, see https://docs.python.org/3/library/typing.html#annotating-generators-and-coroutines

> Simple generators that only ever yield values can also be annotated as having a return type of either [Iterable[YieldType]](https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable) or [Iterator[YieldType]](https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterator):

Finally, if you *must* use `Any`, consider using `object` instead as bottom type, as it doesn't leak into all dependent types, which `Any` does.